projects/ngneat/forms-manager/src/lib/forms-manager.ts

```diff
@@ -374,10 +376,10 @@ export class NgFormsManager<FormsState = any> {
     }
 
     const unsubscribe = merge(
-      control.valueChanges,
-      control.statusChanges.pipe(distinctUntilChanged())
+      control.statusChanges.pipe(distinctUntilChanged()),
+      ...this.getValueChangeStreams(control)
     )
-      .pipe(debounceTime(mergedConfig.debounceTime))
+      .pipe(debounce(value => (value === NO_DEBOUNCE ? EMPTY : timer(mergedConfig.debounceTime))))
```

Comment:
It skips the debounce if a control with updateOn=blur emits a value which is mapped to NO_DEBOUNCE in the getValueChangesStreams() function.

Reply:
Can you explain what's caused the issue? I mean, why the debounce conflicts with the updateOn option?

Reply:
The video I put here shows the problem https://github.com/ngneat/forms-manager/issues/17#issuecomment-678829549 

User types in the updateOn=blur FormControl, then clicks submit, without clicking outside the input box so no blur event is triggered until clicking submit. In the submit handler function then when trying to get the value `this.manager.getControl('stepOne','textarea').value` this still has the old value because the debounce (300ms default) pauses the stream and doesn't hit the update functions in the subscribe. So in this case we shouldn't debounce.

I'm happy to have a zoom(or other) chat to show you the problem in action and discuss further.

Thanks for taking the time to review!